src/chapters/loops/while-loops.rst

```diff
@@ -0,0 +1,199 @@
+   pair: input; validation
+
+This example illustrates the additional flexibility provided by ``while`` loops. While we use ``for`` loops to iterate over fixed collections (a string, an array, a collection of integers), the ``while`` loop can be used to iterate in more general circumstances. For the input validation example, at runtime it cannot be determined how many times the loop will repeat.
+
+.. warning:: It is *technically* possible to write a ``while`` loop as a ``for`` loop, since ``for`` loops use a condition, and the initial and update expressions can be left empty. Doing so, however, results in confusing code. This should not be done under any circumstances!
+
+   .. sourcecode:: js
+
+      const input = require('readline-sync');
+
+      let num = input.question('Please enter a positive number:');
+      num = Number(num);
+
+      for (; num <= 0; ) {
+         num = input.question('Invalid input. Please enter a positive number:');
+         num = Number(num);
+      }
+
+Infinite Loops, Revisited
+-------------------------
+
+.. index::
+   single: loop; infinite
+
+It is easier to create an infinite ``while`` loop than an infinite ``for`` loop. To see this, consider our first ``while`` loop example, but where we forget to update the loop variable.
+
+.. sourcecode:: js
+
+   let i = 0;
+
+   while (i < 51) {
+      console.log(i);
+   }
+
+This is an infinite loop. The variable ``i`` is initialized to 0 never updated, so the condition ``i < 51`` will always be true.
+
+Terminating a Loop With **break**
+---------------------------------
+
+.. index:: 
+   single: ! break
+   single: loop; termination
+
+JavaScript, like most programming languages, provides a mechanism for terminating a loop before it would complete otherwise. The ``break`` keyword, when used within a loop, will immediately terminate the execution of any loop. Program execution then continues at the next line of code below the loop.
+
+.. admonition:: Example
+
+   This loop executes 12 times, for values of ``i`` from 0 to 11. During the twelfth iteration, ``i`` is 11 and the condition ``i > 10`` evaluates to ``true`` for the first time and execution reaches the ``break`` statement. The loop is immediately terminated at that point.
+
+   .. sourcecode:: js
+
+      for (let i = 0; i < 42; i++) {
+         
+         // rest of loop body
+
+         if (i > 10) {
+            break;
+         }
+
+      }
+
+While not used often, ``break`` can be quite helpful in some situations. Consider a situation where we are searching for a particular element in an array. (While we have seen that JavaScript has array methods that can carry out array searches, many programming languages do not.) 
+
+We can use a ``while`` loop to say, essentially, *while we have not reached the end of the array, continue iterating*. We can then include a ``break`` within a conditional check to say, *when we have found the element we are searching for, exit the loop*.
+
+.. sourcecode:: js
```

Comment:
Confusing shift to a ``while`` loop.  The same action could also be done with a ``for`` loop.  Show examples of each, or show different uses of ``break`` - one for each type of loop.

Reply:
OK.